### DISCUSSION TITLE
Anybody got this working with a MusicCast Bar 400?

### DISCUSSION THREAD

**Discussion by brentk7** [mvdwetering/yamaha_ynca#360] [General]
I tried to get this working with a MusicCast Bar 400, but get the following error when trying to add the device 
> Failed to connect, check settings and make sure this is the _only_ application connecting to the receiver with the YNCA protocol.
I have made sure there is nothing else connected.

I also have an RX-V685 and that works fine. Both the Bar 400 and the RXV-685 work with the same Yamaha apps (AV Controller and MuscCast Controller), so I thought the bar should work.
The bar also responds to the web API requests as shown below (exactly the same as the RX-V685)
`http://x.x.x.x/YamahaExtendedControl/v1/system/getDeviceInfo`
```
{"response_code":0,"model_name":"YAS-408","destination":"A","device_id":"F08620B84520","system_id":"02859E73","system_version":5.07,"api_version":2.15,"netmodule_generation":2,"netmodule_version":"1810    ","netmodule_checksum":"776566B4","serial_number":"___redacted___","category_code":2,"operation_mode":"normal","update_error_code":"00000000","update_data_type":0,"analytics_info":{"uuid":"ecd81e24-3752-4389-ab51-4fface962c71"}}
```

Any suggestions on whether the above means this should work and / or what else I can try or any further information I can provide to assist with getting this working?

**Comment by mvdwetering** [comment]
This integration uses the YNCA protocol which is different from the http/webbased protocols like used by the apps.

The error you are getting basically means that the IP address exists, but the port does not accept connections. This happens when the protocol is not supported at all or when something is already connected to that port. I might update that error a bit to also include that the protocol might not be supported.

The modelid of the MusicCast Bar 400 seems to be YAS-408. A quick search for YAS (and YAS-408) in combination with YNCA brings up basically nothing. So it seems the YNCA protocol is not supported by soundbars, only on AV receivers.

**Reply by brentk7** [reply]
Thanks for the response. Looks like I'll have to get the functionality I want using the web protocols sending requests manually.
